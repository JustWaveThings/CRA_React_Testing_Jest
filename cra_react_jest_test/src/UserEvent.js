import React, { useState } from "react";

function UserEvent() {
  const [heading, setHeading] = useState("Test Jest");

  const clickHandler = () => {
    setHeading("Not Testing Jest");
  };

  return (
    <>
      <button type="button" onClick={clickHandler}>
        Click me
      </button>
      <h1>{heading}</h1>
    </>
  );
}

export default UserEvent;
