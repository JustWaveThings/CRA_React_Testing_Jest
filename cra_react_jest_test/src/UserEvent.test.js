import React from "react";
import { render, screen } from "@testing-library/react";
import { act } from "react-dom/test-utils";
import userEvent from "@testing-library/user-event";
import UserEvent from "./UserEvent";

describe("App component", () => {
  it("renders test jest heading", () => {
    const { container } = render(<UserEvent />);
    expect(container).toMatchSnapshot();
  });
  it('renders "not testing jest" heading when button is clicked', async () => {
    const user = userEvent.setup();
    render(<UserEvent />);
    const button = screen.getByRole("button", { name: /click me/i });

    await act(async () => {
      await user.click(button);
    });

    expect(screen.getByRole("heading").textContent).toMatch(
      /Not Testing Jest/i
    );
  });
});
