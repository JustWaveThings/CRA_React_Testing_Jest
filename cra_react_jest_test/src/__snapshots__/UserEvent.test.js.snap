// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders test jest heading 1`] = `
<div>
  <button
    type="button"
  >
    Click me
  </button>
  <h1>
    Test Jest
  </h1>
</div>
`;
